### ISSUE TITLE
Fileselect Allways needs dobble klick

### ISSUE THREAD

**Issue by marvin1099** [xournalpp/xournalpp#5460]
### Operating System

Linux

### (Linux only) Distribution

Arch (Garuda)

### (Linux only) Desktop Environment

KDE

### (Linux Only) Display Server

X.Org

### Installation Method

Arch Repo (extra)

### Xournal++ Version

1.2.2 (09ae3333)

### libgtk Version

3.24.39

### Bug Description

I wold be great if xournal would support single click on the fileselector so its easyer to use on touchscreen.
I have setup my kde to open files and folders in single klick (I like that more anyway) but xournal doesn't seem to have support.
  

### Expected Behaviour

If you have enabled Singleklick in your settings, I would be nice if xournal would open folders on Singleklick.

### Steps to Reproduce

1. Enable single klick in system settings
2. Try to save any file
3. Folders will only open after dobble click

**Comment by marvin1099** [comment]
Im currius, why are you using the gdk filepicker, is it not possible to use system defaut (i have seen something about "xdg-desktop-portal", might not be correct).
Im not shure if it is hard to use the sytem default, i know it is somewath simple on windows, but not shure about linux.

**Comment by rolandlo** [comment]
Hmm, maybe I closed the issue prematurely. We could use the [Gtk.FileChooserNative](https://docs.gtk.org/gtk3/class.FileChooserNative.html) instead. It does offer less features and we may need some of those (not sure). One would have to check if that one supports single click when it's the system's default.
The plugin API (via `app.getFilePath` or `app.saveAs`) uses the native dialog. You could check there if single click is supported.

**Comment by marvin1099** [comment]
So i have tested a python script that uses [Gtk.FileChooserNative](https://docs.gtk.org/gtk3/class.FileChooserNative.html) and it works single click (it just runs the default as expected so works perfectly).
Here is the code i used for testing (It probably wont be much use, but it cant hurt):

    #!/usr/bin/env python3
    
    import gi
    gi.require_version('Gtk', '3.0')
    from gi.repository import Gtk
    
    class FileChooserExample(Gtk.Window):
    
        def __init__(self):
            Gtk.Window.__init__(self, title="FileChooser Example")
            self.set_border_width(10)
    
            button = Gtk.Button(label="Select a file")
            button.connect("clicked", self.on_file_clicked)
    
            self.add(button)
    
        def on_file_clicked(self, widget):
            dialog = Gtk.FileChooserNative(title="Select a file",
                                           action=Gtk.FileChooserAction.OPEN)
            dialog.set_transient_for(self)
    
            response = dialog.run()
            if response == Gtk.ResponseType.ACCEPT:
                print("File selected: " + dialog.get_filename())
            elif response == Gtk.ResponseType.CANCEL:
                print("Canceled file selecttion")
    
            dialog.destroy()
    
    win = FileChooserExample()
    win.connect("destroy", Gtk.main_quit)
    win.show_all()
    Gtk.main()

So when i run the app like this

    python ./Test.py

it reports

    File selected: /home/marvin/Downloads/pras-first-h.png

Expected behavior overall.
I'm not sure what of [GtkFileChooser](https://docs.gtk.org/gtk3/iface.FileChooser.html) you are using, but if the app is just grabbing the path then this should work.

It would be great if it was implemented.
For saving and loading of course, that would make the file selection so much easyer.

**Comment by marvin1099** [comment]
> Hmm, maybe I closed the issue prematurely. We could use the [Gtk.FileChooserNative](https://docs.gtk.org/gtk3/class.FileChooserNative.html) instead. It does offer less features and we may need some of those (not sure). One would have to check if that one supports single click when it's the system's default. The plugin API (via `app.getFilePath` or `app.saveAs`) uses the native dialog. You could check there if single click is supported.

Please consider reopening the issue,
as a alternative title it could also say "Request to change GtkFileChooser to GtkFileChooserNative".
If you still don't think this should be something that is part of the project, than say so and i put a pin on it.
If you don't answer in like a weak, i will just open a new issue, to get this reconsidered.
Thanks for all comments and input.

**Comment by marvin1099** [comment]
Thank you.
Hopefully it gets implemented, I have my fingers crossed.

**Comment by tockudex** [comment]
> Thank you. Hopefully it gets implemented at some point, I have my fingers crossed.

If you're still interested in the implementation of this feature, I believe a pull request with the required modification could greatly accelerate said implementation.

**Comment by marvin1099** [comment]
> > Thank you. Hopefully it gets implemented at some point, I have my fingers crossed.
> 
> If you're still interested in the implementation of this feature, I believe a pull request with the required modification could greatly accelerate said implementation.

Yes I'm but I'm not familiar with the codebase and don't have a lot of c experience.
So I max be able to figure it out bit I'm not shure jet.
Would still be great to have,
i will try to see if I can find the relevant code parts in the next weaks and try to make to work.

**Comment by tockudex** [comment]
Glad to hear it, thanks !
I'd like to help, but I have literary no experience in C or C++…

**Comment by marvin1099** [comment]
Jep i dont think this will be added any time soon.
I just looked at the codebase and im smart enoght to know when something is above me.
This is in fact way above me.
I only code a bit in my free time and code with python usually
and even then i do simple cli scripts.

I still hope someone else will try to figure this out,
but I just have basic c++ experience.
I can't make this happen if someone else looks at this
and might be able to swich out [GtkFileChooser](https://docs.gtk.org/gtk3/iface.FileChooser.html) for [Gtk.FileChooserNative](https://docs.gtk.org/gtk3/class.FileChooserNative.html) please try to do so.
I would be very grateful.

**Comment by LDprg** [comment]
I would also like to see this implemented. Most apps nowadays work correctly with the xdg-portals, and it is very annoying that xournalpp does not.

**Comment by tockudex** [comment]
For proper triaging, the label “bug” should be replaced by “enhancement”, “os::linux” and “UX”.

Could you take a look at this please, @rolandlo as I'm not sure @marvin1099 can apply the changes himself?

**Comment by LDprg** [comment]
I did create a PR solving this, which should mostly work (I did notice one bug and I did not double-check to really replace all file dialogs). 

However, my testing is limited to my main system (KDE Wayland, Arch btw.). So feel free to give feedback.

**Comment by marvin1099** [comment]
Thanks that you took the time to make this @LDprg , I did not think that anyone would make a implementation of this.

As for feedback I may tr y to compile and run soon and will report at that point.

Also thanks @tockudex for suggesting the tag change and @rolandlo for adding the correct tags.

I think this makes this suggestion more transparent and therefore, I would argue have a higher interest that before.

Might not be true but can't be worse than before where there was no interest on this.

**Comment by marvin1099** [comment]
@rolandlo How about the implementation of @LDprg ?
Is it usable?
I would still like this feature.

**Comment by tockudex** [comment]
@marvin1099 the development of this feature in the PR #6049 has stalled, so not usable yet.

**Comment by rolandlo** [comment]
> [@rolandlo](https://github.com/rolandlo) How about the implementation of [@LDprg](https://github.com/LDprg) ? Is it usable? I would still like this feature.

There is a segfault (crash) left to fix in @LDprg 's approach https://github.com/xournalpp/xournalpp/pull/6049#issuecomment-2486693884, which requires some refactoring (and also rebasing on current master).

**Comment by marvin1099** [comment]
Ok, still a good amount of work to do is what you're saying.
And since it has been a while its behind so also some rebasing, man.
Sad, probably won't be coming soon then.
Well thanks for the update.

**Comment by LDprg** [comment]
Yeah sorry there are still some bugs and incomplete code parts. Sadly I won't get to finish it anytime soon (even though I would still like this feature myself), since I am currently very busy.

If anyone wants to continue work feel free to do so.
